fala/apps/adviser/views.py

```diff
@@ -43,33 +42,84 @@ class AccessibilityView(TemplateView):
+                self._data = self._form.search()
+            else:
+                self._data = {}
+        else:
+            self._data = {}
         return super().get(self, request, *args, **kwargs)
 
     # Errors from LaaLaaAPI manifest as _data being falsey, so we bounce back to search page
     def get_template_names(self):
-        if self._data:
-            template = "adviser/results.html"
+        if self._form.is_valid():
+            if self._region == Region.ENGLAND_OR_WALES:
+                template = "adviser/results.html"
+            else:
+                template = "adviser/other_region.html"
         else:
             template = "adviser/search.html"
+
         return [template]
 
     def get_queryset(self):
         return self._data.get("results", None)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context.update({"form": self._form, "data": self._data})
-        # Only paginate on success. _data is already paginated, so we need our own paginator
-        # that can answer questions about next and previous buttons etc.
-        if self._data:
-            pages = LaaLaaPaginator(self._data["count"], 10, 3, self._form.current_page)
-            params = {
-                "postcode": self._form.cleaned_data["postcode"],
-                "name": self._form.cleaned_data["name"],
-            }
-            # create list of 'categories=x' so that they can be referenced in URL links
-            categories = list(map(lambda c: "categories=" + c, self._form.cleaned_data["categories"]))
 
-            context.update({"pages": pages, "params": params, "categories": categories})
+        if self._form.is_valid():
+            if self._region == Region.ENGLAND_OR_WALES:
+                pages = LaaLaaPaginator(self._data["count"], 10, 3, self._form.current_page)
+                params = {
+                    "postcode": self._form.cleaned_data["postcode"],
+                    "name": self._form.cleaned_data["name"],
+                }
+                categories = list(map(lambda c: "categories=" + c, self._form.cleaned_data["categories"]))
+                context.update(
+                    {
+                        "form": self._form,
+                        "data": self._data,
+                        "pages": pages,
+                        "params": params,
+                        "categories": categories,
+                    }
+                )
+            else:
+                region_data = REGION_TO_LINK[self._region]
+                context.update(
+                    {
+                        "postcode": self._form.cleaned_data["postcode"],
+                        "link": region_data["link"],
+                        "region": region_data["region"],
+                    }
+                )
+        else:
+            context.update({"form": self._form, "data": self._data})
         return context
+
+
+REGION_TO_LINK = {
```

Comment:
Weird place for this

Reply:
I almost feel this should be a JSON file on it's own as a resource that we import in and use.

Reply:
Moved into the view scope

Reply:
...and now into the 'OtherJurisdiction' state